### ISSUE TITLE
Problems with estimating rho

### ISSUE THREAD

**Issue by lningyawen** [CshlSiepelLab/phast#44]
Hello Ritika

I estimated rho (expected substitution rate of conserved elements relative to neutrality) using phastCons, and run separately on non-overlapping 1 MB chunks of the input alignment. However,  I found that some of the chunks were not producing results (cons.mod and noncons.mod) when they were calculated, but kept calculating and  producing the output shown below.
```
Reading tree model from ../../ave.cons.mod...
Reading alignment from chr15.10000504-10999289.ss...
Creating 'conserved' and 'nonconserved' states in HMM...
Computing emission probs (state 0, cat 0, mod 0)...
Computing emission probs (state 1, cat 1, mod 1)...
Finding MLE for (mu, nu, rho)...
WARNING: likelihood decreased during EM: it 51 total_logl=-12643913.18, prev_total_logl=-12643909.22
WARNING: likelihood decreased during EM: it 56 total_logl=-12643881.39, prev_total_logl=-12643875.4
WARNING: likelihood decreased during EM: it 61 total_logl=-12643857.79, prev_total_logl=-12643851.8
..........................................................................................................................................................................
..........................................................................................................................................................................
..........................................................................................................................................................................
WARNING: likelihood decreased during EM: it 147 total_logl=-12643811.32, prev_total_logl=-12643806.49
WARNING: likelihood decreased during EM: it 149 total_logl=-12643812.07, prev_total_logl=-12643805.77
```

At the same time, there are some chunks that get results within a few hours of being calculated and their output is shown below.
```
Reading tree model from ../../ave.cons.mod...
Reading alignment from chr13.90000001-91000000.ss...
Creating 'conserved' and 'nonconserved' states in HMM...
Computing emission probs (state 0, cat 0, mod 0)...
Computing emission probs (state 1, cat 1, mod 1)...
Finding MLE for (mu, nu, rho)...
WARNING: likelihood decreased during EM: it 47 total_logl=-13999699.55, prev_total_logl=-13999698.22
WARNING: likelihood decreased during EM: it 55 total_logl=-13999656.36, prev_total_logl=-13999656.19
WARNING: likelihood decreased during EM: it 57 total_logl=-13999658.18, prev_total_logl=-13999651.68
WARNING: likelihood decreased during EM: it 61 total_logl=-13999654.41, prev_total_logl=-13999646.09
WARNING: likelihood decreased during EM: it 63 total_logl=-13999647.39, prev_total_logl=-13999641.86
WARNING: likelihood decreased during EM: it 65 total_logl=-13999644.04, prev_total_logl=-13999643.6
(mu = 0.158472. nu = 0.205849, rho = 0.316591)
Writing re-estimated tree models to chr13.90000001-91000000.ss.cons.mod and chr13.90000001-91000000.ss.noncons.mod...
Done.
```
As the output above shows, tasks that go wrong can generate many, many `WARNING: likelihood decreased during EM: it..............`. The command I used is `ls *.ss | while read ss;do /pwd/bin/phastCons --estimate-rho $ss --no-post-probs $ss ../../ave.cons.mod &> ${ss}.conserved_model.log; done;`

Thank you for any help you can provide.

Best,
Yawen

**Comment by siepellab** [comment]
Hello,
The difference in the likelihood is pretty small so in this case I wouldn't worry about it.

Hope this helps!
Ritika

**Comment by lningyawen** [comment]
Hello Ritika,

Sorry for getting back to you late.

I would like to say that when calculating this command `ls *.ss | while read ss;do /pwd/bin/phastCons --estimate-rho $ss --no-post-probs $ss ../../ave.cons.mod &> ${ss}.conserved_model.log; done`,  there were some `*.ss` files that ran very quickly and got the results, while others run for days with no results, except for a lot of` WARNING: likelihood decreased during EM: it 47 total_logl=-13999699.55, prev_total_logl=-13999698.22` in the log. 

However, these files are about the same size, about 50 ~ 60 MB. And these `*.ss`  files were calculated from the command  `msa_split pwd/${i}.maf  --in-format MAF  --windows 1000000,0  --out-root ./split_MSA/$i/$file/  --out-format SS   --min-informative 1000   -between-blocks 5000`.

Thanks for your helps!
Best, Yawen

**Comment by siepellab** [comment]
Hello, 

So there are no error messages in the log? I would start by making sure that your MAF files are split by chromosome of the reference species and are sorted by the reference genome. This is something all PHAST tools expect. Doesn't seem like that is the issue here but always good to check. You can also try to run it on larger chunks in case there isn't enough
data. 
If this still doesn't work would you be able to send us a smaller subset of the files so that we can replicate it on our end. 

Thanks!
Ritika

**Comment by lningyawen** [comment]
Hello Ritika,

There are no error messages in my log file, except for a large number of` "WARNING: likelihood decreased during EM: it 47 total_logl=-13999699.55, prev_total_logl=-13999698.22 "`. I double checked my MAF file and it was split by reference 
 chromosome and sorted by genome.

Also, I found that my command is `ls *.ss | while read ss;do /gpfs/home/liunyw/biosoft/phast/bin/phastCons --estimate-rho $ss --no-post-probs $ss . /... /ave.cons.mod &> ${ss}.conserved_model.log`, so when one chunk fails to compute (encountering the situation we talked about above), it affects the later chunks that fail to compute. When I remove this uncomputable chunk, and resubmit the command again, it continues to run. In this case, when I encounter a problem that can't be calculated, I just remove the chunks.  And how do I send one of these uncomputable chunks  to you?

Besides, You mentioned running it on a larger chunk of data, is it possible to set the `--windows` parameter to a larger size when splitting the maf file, e.g. `--windows 1500000,0`.


Thank you very much!
Best, Yawen
